benchmark/url/whatwg-url-to-and-from-path.js

```diff
@@ -3,28 +3,50 @@ const common = require('../common.js');
 const { fileURLToPath, pathToFileURL } = require('node:url');
 const isWindows = process.platform === 'win32';
 
-const bench = common.createBenchmark(main, {
-  input: isWindows ? [
-    'file:///c/',
-  ] : [
-    'file:///dev/null',
-    'file:///dev/null?key=param&bool',
-    'file:///dev/null?key=param&bool#hash',
-  ],
-  method: isWindows ? [
-    'fileURLToPath',
-  ] : [
-    'fileURLToPath',
-    'pathToFileURL',
-  ],
-  n: [5e6],
-});
+const inputs = isWindows ? [
+  'C:\\foo',
+  'C:\\Program Files\\Music\\Web Sys\\main.html?REQUEST=RADIO',
+  '\\\\nas\\My Docs\\File.doc',
+  'file:///C:/foo',
+  'file:///C:/dir/foo?query=1',
+  'file:///C:/dir/foo#fragment',
+] : [
+  '/dev/null',
+  '/dev/null?key=param&bool',
+  '/dev/null?key=param&bool#hash',
+  'file:///dev/null',
+  'file:///dev/null?key=param&bool',
+  'file:///dev/null?key=param&bool#hash',
+];
 
-function main({ n, input, method }) {
-  method = method === 'fileURLOrPath' ? fileURLToPath : pathToFileURL;
+const bench = common.createBenchmark(
+  main,
+  {
+    method: ['pathToFileURL', 'fileURLToPath'],
+    input: Object.values(inputs),
+    n: [5e6],
+  },
+  {
+    combinationFilter: (p) => {
+      if (!isWindows) {
+        return (
+          (p.method === 'pathToFileURL') ||
+          (p.input.startsWith('file://') && p.method === 'fileURLToPath')
+        );
+      }
+      return (
+        (!p.input.startsWith('file://') && p.method === 'pathToFileURL') ||
+        (p.input.startsWith('file://') && (!isWindows || p.method === 'fileURLToPath'))
```

Comment:
`isWindows` is guaranteed to be falsy in this branch
```suggestion
        (p.input.startsWith('file://') &&  || p.method === 'fileURLToPath')
```